src/com/amazon/ion/IonBufferConfiguration.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.ion;
+
+/**
+ * Configures Ion lookahead buffers.
+ */
+public final class IonBufferConfiguration extends BufferConfiguration<IonBufferEventHandler, IonBufferConfiguration> {
+
+    /**
+     * Builds IonBufferConfiguration instances.
+     */
+    public static final class Builder
+        extends BufferConfiguration.Builder<IonBufferEventHandler, IonBufferConfiguration, Builder> {
+
+        /**
+         * 4-byte IVM + 1 byte user value.
+         */
+        private static final int MINIMUM_MAX_VALUE_SIZE = 5;
+
+        /**
+         * An IonBufferEventHandler that does nothing.
+         */
+        private static final IonBufferEventHandler NO_OP_EVENT_HANDLER = new IonBufferEventHandler() {
+
+            @Override
+            public void onOversizedSymbolTable() {
+                // Do nothing.
```

Comment:
Especially given that the docs for this method say:

>  Because symbol tables cannot be truncated without corrupting values that follow in the stream, this condition is not recoverable.

I think I'd expect the default handler for this to throw an exception.

Reply:
Here's why I didn't do this.

1. A symbol table can only be oversized if a maximum buffer size is provided.
2. If a maximum buffer size is provided, an error is raised if a handler is not provided.
3. Therefore, this no-op handler will not be used by default when it's possible to encounter an oversized symbol table. A user would have to explicitly pass `builder.getNoOpBufferEventHandler()` to `builder.withHandler` on a builder with a maximum value specified.
4. This is no different than the user providing a custom implementation of IonBufferEventHandler that does nothing in its `onOversizedSymbolTable` override. Since we don't know what the user's override does, we need to be able to protect against this by aborting processing after `onOversizedSymbolTable` returns, if it ever does. This is done [here](https://github.com/amzn/ion-java/blob/b577e0a5d1b4910f8e5e7080def5bcc3e8ba125f/src/com/amazon/ion/impl/IonReaderLookaheadBuffer.java#L746).

Reply:
Makes sense! In this case, I was misled by the combination of the name "default handler" (which says to me "this is what will happen unless a user specifies otherwise") and the comment "Do nothing." In reality, this method is never called, so "do nothing" never happens. (Whoa, man.)

Consider adding a terse version of your explanation to the comment?

```suggestion
                // This implementation will never be called.
                // If no maximum buffer size is configured, symbol tables cannot be considered oversized.
                // If a maximum buffer size is configured, an event handler that overrides this 
                // implementation will also be configured.
```

or even more tersely:

```suggestion
                // This implementation will never be called.
```

Though the latter does omit the loophole you mentioned:
> A user would have to explicitly pass builder.getNoOpBufferEventHandler() to builder.withHandler on a builder with a maximum value specified.

Reply:
I ended up with
```
// If no maximum buffer size is configured, symbol tables cannot be considered oversized and this
// implementation will never be called.
// If a maximum buffer size is configured, a handler must also be configured. In that case,
// this implementation will only be called if the user provides it to the builder manually. 
```